pandas/core/dtypes/cast.py

```diff
@@ -312,12 +314,11 @@ def maybe_cast_result_dtype(dtype: DtypeObj, how: str) -> DtypeObj:
     DtypeObj
         The desired dtype of the result.
     """
-    d = {
-        (np.dtype(np.bool), "add"): np.dtype(np.int64),
-        (np.dtype(np.bool), "cumsum"): np.dtype(np.int64),
-        (np.dtype(np.bool), "sum"): np.dtype(np.int64),
-    }
-    return d.get((dtype, how), dtype)
+    if how in ["add", "cumsum", "sum"] and (dtype == np.dtype(np.bool)):
```

Comment:
use
```issubclass(dtype, np.bool_)```

Reply:
This ends up raising in a test:
```python
E       TypeError: issubclass() arg 1 must be a class
```
I think for np.dtype("float64") as the input

Reply:
use type(dtype)

Reply:
or i guess its ok as is

Reply:
though maybe

```np.dtype(dtype) == "bool"``` is better

Reply:
depends on what we allow dtype at this point